### ISSUE TITLE
Compatibility check does not point out source of incompatibility in case of missing/disabled webaudio API

### ISSUE THREAD

**Issue by mpeter50** [element-hq/element-web#28153]
### Steps to reproduce

1. I'm starting with a Firefox-based browser with no webpages open. It's `dom.webaudio.enabled` about:config settings has happened to be false
2. I have navigated to https://app.element.io/
3. The webapp tells there is a compatibility issue with my browser, but does not tell what is that
4. I click the "Continue anyway" button
5. The webapp shows the unsupported browser page, but does not tell why is that

#### What did you expect?
A short (or longer, when more information would be useful) description that Element could not access the browser's WebAudio API. An error message you got could also be useful to be displayed for the generic case.

#### What happened instead?
The webapp did not tell the reason of incompatibility.

#### Additional information

According to #27864, you have already implemented such a feature, but it seems there may be something preventing it from working.
I have mentioned the issue there earlier, but did not get a response about whether this is expected or unexpected behavior, so I have opened this issue in the belief that this is unexpected behavior.

I expect that the incompatibility in this case is the missing WebAudio API, because if I enable it in about:config the webapp will load fine.

### Operating system

Windows

### Browser information

Firefox

### URL for webapp

https://app.element.io/

### Will you send logs?

No

**Comment by mpeter50** [comment]
Screenshots about how the webapp looks like at the 3rd and 5th points:

|Compatibility warning page|Failure page|
|-|-|
|![image](https://github.com/user-attachments/assets/519ee3aa-8fd3-40a1-b89d-d374cf25f196)|![image](https://github.com/user-attachments/assets/598a7796-e719-48fb-95a5-1a081be36c13)|

**Comment by t3chguy** [comment]
The designers explicitly did not want any UI to show exactly what feature is missing, nor can we realistically get translations for all the various things which can be missing.

**Comment by mpeter50** [comment]
> The designers explicitly did not want any UI to show exactly what feature is missing

What is the reason behind that? Would it help if this information was only shown in a default-collapsed menu?

I think it would be useful to have this because it can be actionable by the user.
Incompatibility can not only be caused by old software, but also by disabled features, like this one. For example, if the user has disabled webaudio support because of the fingerprinting possibility on any website, with the understanding that they are fine without web calls, seeing that it causes a problem here can make them to consider to switch it on again, and to approach this fingerprintability problem in a different way.

This does not only apply to the current webaudio api thing, but generally to other incompatibility cases too.

> nor can we realistically get translations for all the various things which can be missing.

I see. Do we need translations, though? I would expect that if the user does not immediately understand what "WebAudio API is not available" means and what to do to fix it, they would need to reach for external help to resolve such an issue anyway.
What I wanted to say is that I think it may not be useful to translate this, kind of similarly to log messages.

This way, the added value of this would be that the problem is unambiguous, and the user does not need to dive in the logs, where several other warnings, maybe error level messages too are visible. The external help would also not need to try to get the user to look at the logs, and forward them somehow, either.

**Comment by t3chguy** [comment]
> I see. Do we need translations, though?

Yes, multiple customers require translatability for all parts of the app

**Comment by grinapo** [comment]
The usual solution is to have a non-intrusive button or link saying "technical information for developers" which does not require inner translation, especially if it spews obvsiously tech-looking things like a json. Except it would not be _you_ developers but all of _us_ out here. 

The main point is not to inform the user but to offer the possibility for any IT people to help the user to determine what the problem may be. Maybe even you, developers, in the end.

(Funny that I wanted to say that maybe debug logs contain that information but realised that "Download logs" does exactly nothing, and definitely not downloading debug logs. Puzzling. So, I may mean a working debug logs output or even better an excerpt regarding the specific infompatibilities.)

**Comment by grinapo** [comment]
> sounds like an issue on your end

Issued separately, thank you. Please do not let that to distract you from this original issue.

**Comment by MarcWadai** [comment]
There also might be other browser settings that cause the webaudio to not work.
Will it be possible to add a check like `new RTCPeerConnection()` to see if it is even possible to start a call on the current user browser settings ?

**Comment by K4sum1** [comment]
Why does disabling webaudio cause element to break? Why does the "Continue anyway" button not actually fucking continue anyway? It just goes to another error without any way to bypass it. I spent ages trying to figure out how to block rageshake.ts since it doesn't show in the network tab for whatever fucking reason before I noticed it seemed audio related and enabling webaudio got past the error. I don't care for audio or voice or anything. I just want an easy basic chat app since nheko decided it no longer wants to connect and I can't be bothered to diagnose why yet.

**Comment by t3chguy** [comment]
Doesn't the continue anyway screen warn you that it may not work? Web audio is used to play notification sounds and the code doesn't run conditionally, contributions welcome, just as the webaudio stuff was community contributed.

**Comment by K4sum1** [comment]
> Doesn't the continue anyway screen warn you that it may not work?

Yes, but I'd expect audio or something to be broken, not just an artificial error message with no bypass. Even if it was something major, I'd rather see a half broken UI than an error of your browser doesn't work, fuck you lol.
